.github/workflows/test.yml

```diff
@@ -82,3 +88,64 @@ jobs:
         if: ${{ !startsWith(matrix.go, '1.18.') && !startsWith(matrix.go, '1.19.') }}
         run: |
           go test -race -shuffle=on -v -count=10 ./...
+
+  freebsd:
```

Comment:
Can we integrate this with the above?

```
    strategy:
      matrix:
        os: [linux, macos, windows, freebsd]
        go: ['1.18.x', '1.19.x', '1.20.x', '1.21.x']
    runs-on: ${{ matrix.os == 'linux' ? 'ubuntu-latest' : (matrix.os == 'macos' ? 'macos-latest' : ( matrix.os == 'windows' ? 'windows-latest' : 'macos-12')) }}
```

For each actions, you can use `if`:

```
- name: build in FreeBSD
   if: matrix.os == 'freebsd'
   run: |
```

Reply:
setup-go doesn't support freebsd, so there's only 1 go matrix possible, whatever is installed via `pkg add -y lang/go` in freebsd (which is 1.20 at the moment).

Reply:
You can make a special set up section with `if`, right?

```
- name: Setup Go
   if: ${{.matrix.os != 'freebsd' }}
   ...
- name: Setup Go (FreeBSD)
   if ${{ matrix.os == 'freebsd'}}
```

> pkg add -y lang/go in freebsd (which is 1.20 at the moment).

Can't we use the official Go binaries for various Go version?

Reply:
they mentioned freebsd not supported: https://github.com/actions/setup-go/blob/main/src/system.ts#L5

Reply:
I mean whether we can use https://go.dev/dl/

```
curl https://go.dev/dl/go1.20.7.freebsd-amd64.tar.gz
rm -rf /usr/local/go && tar -C /usr/local -xzf go1.21.0.freebsd-amd64.tar.gz
```